htlcswitch/link.go

```diff
@@ -2036,6 +2036,20 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			}
 		}
 
+		// Lookup the original htlc, if it was part of a blinded route
+		// switch out our error.
+		blindingFailure, err := l.blindedRouteFailure(msg.ID)
```

Comment:
so just checking my understanding: if our peer is behaving correctly, the case lnwire.CodeInvalidBlinding: case above will always be overwritten here right? cause they only send Malformed back if we are also part of the path

Reply:
> if our peer is behaving correctly, the case lnwire.CodeInvalidBlinding: case above will always be overwritten here

Yeah. Didn't want to tie the blinded conversion to getting the `Malformed` (in case of buggy/bad peers) so I think this is the simplest way of handling it?